runtime_lib/CMakeLists.txt

```diff
@@ -128,148 +128,21 @@ foreach(target ${AIE_RUNTIME_TARGETS})
-    imageheadertable.cpp
-    imageheadertable-versal.cpp
-    imageheadertable-zynq.cpp
-    imageheadertable-zynqmp.cpp
-    kdf.cpp
-    Keccak-compact.cpp
-    Keccak-compact-versal.cpp
-    logger.cpp
-    # main.cpp
-    mcsfile.cpp
-    options.cpp
-    outputfile.cpp
-    outputfile-versal.cpp
-    parsing.cpp
-    partition.cpp
-    partitionheadertable.cpp
-    partitionheadertable-versal.cpp
-    partitionheadertable-zynq.cpp
-    partitionheadertable-zynqmp.cpp
-    partition-versal.cpp
-    postprocess.cpp
-    readimage.cpp
-    readimage-versal.cpp
-    readimage-zynq.cpp
-    readimage-zynqmp.cpp
-    reginit.cpp
-    reginit.tab.cpp
-    reginit.yy.cpp
-    verifyimage-versal.cpp
-    verifyimage-zynqmp.cpp
-)
-list(TRANSFORM libsources PREPEND ${BOOTGEN_SOURCE}/)
-
-if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows|Darwin")
-# Windows fails CI because OpenSSL is not installed
-# Macos fails because no malloc.h???
-add_library(Bootgen STATIC ${libsources})
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
-  set(bootgen_warning_ignores
-    -Wno-reorder
-    -Wno-unused-private-field
-    -Wno-deprecated-register
-    -Wno-overloaded-virtual
-    -Wno-deprecated-declarations
-    -Wno-pointer-bool-conversion
-    -Wno-unused-const-variable
-    -Wno-tautological-overlap-compare
-    -Wno-dynamic-class-memaccess
-    -Wno-misleading-indentation
-    -Wno-suggest-override
-    -Wno-non-virtual-dtor
-    -Wno-deprecated-copy
-    -Wno-cast-qual
-    -Wno-date-time
-    -Wno-implicit-fallthrough
-    -Wno-incompatible-pointer-types-discards-qualifiers
-    -Wno-sign-compare
-    -Wno-covered-switch-default
-  )
-else()
-  set(bootgen_warning_ignores)
-endif()
-target_compile_options(Bootgen PRIVATE ${bootgen_warning_ignores})
-target_include_directories(Bootgen PUBLIC ${BOOTGEN_SOURCE})
-
-add_executable(bootgen ${BOOTGEN_SOURCE}/main.cpp)
-target_include_directories(bootgen PUBLIC ${BOOTGEN_SOURCE})
-target_compile_options(bootgen PRIVATE ${bootgen_warning_ignores})
-target_link_libraries(bootgen Bootgen Bootgen ssl crypto)
+include("xaiengine/aiert.cmake")
+set(XAIE_SOURCE_DIR "xaiengine/aie-rt/driver/src")
 
-install(TARGETS bootgen Bootgen)
-endif() #NOT MSVC
-
-if(AIE_ENABLE_GENERATE_CDO_DIRECT)
-  include("xaiengine/aiert.cmake")
-  set(XAIE_SOURCE_DIR "xaiengine/aie-rt/driver/src")
-
-  add_aiert_headers(xaienginecdo_static
-          ${XAIE_SOURCE_DIR}
-          ${AIE_BINARY_DIR}/include
-          ${CMAKE_INSTALL_PREFIX}/include/xaiengine
-  )
-  add_aiert_library(xaienginecdo_static ${XAIE_SOURCE_DIR} STATIC)
+add_aiert_headers(xaienginecdo_static
+        ${XAIE_SOURCE_DIR}
+        ${AIE_BINARY_DIR}/include
+        ${CMAKE_INSTALL_PREFIX}/include/xaiengine
+)
+add_aiert_library(xaienginecdo_static ${XAIE_SOURCE_DIR} STATIC)
 
-  target_link_libraries(xaienginecdo_static PUBLIC cdo_driver)
-  target_compile_definitions(xaienginecdo_static PRIVATE -D__AIECDO__)
+target_link_libraries(xaienginecdo_static PRIVATE cdo_driver)
```

Comment:
need a `target_include_directory(xaienginecdo_static INTERFACE <SOMEPLACE_INSIDE_BOOTGEN>/cdo_driver)` here I believe to get tests to pass.

Reply:
My hope was to avoid having cdo_driver.h exposed to users..  did I miss some place?